### PR TITLE
feat(headless/commerce) add notify trigger controller

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
https://coveord.atlassian.net/browse/KIT-3166

### CODE REVIEW

**Review by louis-bompart — APPROVED** [review]
LGTM, did not review the tests tho.

File: packages/headless/src/controllers/commerce/triggers/headless-commerce-notify-trigger.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import {TriggerSection} from '../../../state/state-sections';
+import {arrayEqual} from '../../../utils/compare-utils';
+import {loadReducerError} from '../../../utils/errors';
+import {buildController} from '../../controller/headless-controller';
+import {NotifyTrigger} from '../../core/triggers/headless-core-notify-trigger';
+
+/**
+ * Creates a `NotifyTrigger` controller instance.
+ *
+ * @param engine - The headless engine.
```
Comment (Spuffynism) [inline packages/headless/src/controllers/commerce/triggers/headless-commerce-notify-trigger.ts:13]:
```suggestion
 * @param engine - The headless commerce engine.
```

File: packages/headless/src/controllers/core/triggers/headless-core-notify-trigger.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {Controller} from '../../controller/headless-controller';
+
+/**
+ * The `NotifyTrigger` controller handles notify triggers. A [Notify trigger] (https://docs.coveo.com/en/3413#notify) query pipeline rule lets you define a message to be displayed to the end user when a certain condition is met.
```
Comment (Spuffynism) [inline packages/headless/src/controllers/core/triggers/headless-core-notify-trigger.ts:4]: I think these need to be next to eachother for the link to work
```suggestion
 * The `NotifyTrigger` controller handles notify triggers. A [Notify trigger](https://docs.coveo.com/en/3413#notify) query pipeline rule lets you define a message to be displayed to the end user when a certain condition is met.
```

**Review by Spuffynism — APPROVED** [review]
nitpicks, looks good!

**Pull request merged**